### PR TITLE
fix(ivy): unable to bind style to zero

### LINKED ISSUE #32984: ivy doesn't handle style value 0 (zero) correctly
# 🐞 bug report

### Affected Package
`@angular/core`

### Is this a regression?

Yes. Works in `~9.0.0-next.9` with Ivy disabled but does not work with Ivy enabled.

### Description

`[style.property]` treats `0` as `'0'` in non-Ivy Angular but as falsy in Ivy which in turn removes the attribute. Since it could be a computed value (`[style.opacity]="1 - somevar"`) it's probably best to treat `0` as an actual value.

## 🔬 Minimal Reproduction

Without Ivy:
- `[style.opacity]="0"` -> applied correctly (`opacity: 0`)
- `[style.opacity]="'0'"` -> applied correctly (`opacity: 0`)

With Ivy:
- `[style.opacity]="0"` -> not applied (no `opacity` set)
- `[style.opacity]="'0'"` -> applied correctly (`opacity: 0`)

## 🌍  Your Environment

**Angular Version:**
```
Angular CLI: 9.0.0-next.7
Node: 12.8.1
OS: darwin x64
Angular: 9.0.0-next.9
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-next.7
@angular-devkit/build-angular     0.900.0-next.7
@angular-devkit/build-optimizer   0.900.0-next.7
@angular-devkit/build-webpack     0.900.0-next.7
@angular-devkit/core              9.0.0-next.7
@angular-devkit/schematics        9.0.0-next.7
@angular/cli                      9.0.0-next.7
@ngtools/webpack                  9.0.0-next.7
@nguniversal/common               9.0.0-next.2
@schematics/angular               9.0.0-next.7
@schematics/update                0.900.0-next.7
rxjs                              7.0.0-alpha.0
typescript                        3.5.3
webpack                           4.41.0
```

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes not being able to bind zero as a value in style bindings.

Fixes #32984.

### CODE REVIEW

File: packages/core/test/acceptance/styling_spec.ts [hunk 1]
```diff
@@ -137,6 +137,20 @@ describe('styling', () => {
     expect(fixture.nativeElement.innerHTML).toBe('<div></div>');
   });
 
+  it('should be able to bind to zero', () => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/styling_spec.ts]: nit: "bind zero" or "bind to zero" ?

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM. 

Wonder why bundle size got up by 27 bytes but I guess this is due to the `isStylingValueDefined` usage...

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](http://test/OCL:272895154:BASE:272895165:1570207497266:1809ed7a)
- [Ivy Presubmit](http://test/OCL:272895154:BASE:272895302:1570207546318:b08052bd)

**Review by mhevery — APPROVED** [review]

**Review by kara — APPROVED** [review]
LGTM

**Review by Splaktar — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
FYI, Ivy presubmit indicated a problem while running one of the targets with the changes from this PR. The problem happens because of the naming collision with `window.ng` when Closure Compiler minifies scripts (i.e. some of the function names were minified to `ng`). This problem is unrelated to this PR, but it just so happens that it triggers the issue. There is a [PR in progress](https://github.com/angular/angular/pull/33009) to resolve the `ng` collision (by adding `ng` to externs list). When the mentioned PR lands, we can proceed with this PR. Thank you.

**Comment by AndrewKushnir** [comment]
FYI, PR #33010 is merged into master now, which should unblock this PR. I ran failing g3 test manually with the fixes from the PR #33010 and everything passes, so no additional presubmits are required and PR is ready to go. Thank you.

**Pull request closed without merging**